File: readability.py
import re


def main():
    # Get text
    text = input('Please Input a Text: ')

    # Split the text into words
    word_count = count_words(text)

    # Count the letters
    letters = count_letters(text)c

    # Split the text into sentences
    sentences = count_sentences(text)

    # Use the Coleman-Liau Index
    # Coleman-Liau Index was designed to tell what U.S. grade level is needed to understand a given text
    # L = Average number of letters per 100 words
    # S = Average number of sentences per 100 words
    # Formula = 0.0588 * L - 0.296 * S - 15.8

    l = 100 * (letters/word_count)
    s = 100 * (sentences/word_count)

    readability = 0.0588 * l - 0.296 * s - 15.8

    # Print Index
    if round(readability) < 1:
        print('Before Grade 1')
    elif round(readability) >= 16:
        print('Grade 16+')
    else:
        print(f'Grade {round(readability)}')


def count_words(text):
    text = text.replace('.', '')
    words = text.split(' ')
    return len(words)


def count_letters(text):
    for ch in ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '"', "'", '.', ', ', '!', '?', '-', '$', '@', '()', ';', ':', ' ', '&', '#', '*']:
        if ch in text:
            text = text.replace(ch, '')
    return len(text)


def count_sentences(text):
    text = text.replace('"', '')
    text = re.split('\.|\?|!', text)
    if not text[-1]:
        text.pop()
    return len(text)


main()

File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{

    string text = get_string("Please input a text: ");

// Count Letters
    int letters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
        else
        {
            letters += 0;
        }
    }

// Count Words
    int words = 0;

    for (int w = 0; w < strlen(text); w++)
    {
        while (isalpha(text[w]) || text[w] == '\'' || text[w] == '-')
        {
            w++;
        }

        while (ispunct(text[w]) || isspace(text[w]))
        {
            w++;
        }

        words += 1;
    }

// Count Sentences

    int sentences = 0;

    for (int s = 0; s < strlen(text); s++)
    {
        if (text[s] == '!' || text[s] == '.' || text[s] == '?')
        {
            sentences += 1;
        }
    }

// Use the Formula

    float l = 100 * (letters / (float) words);
    float s = 100 * (sentences / (float) words);

    //printf("l = %f\n",l);
    //printf("s = %f\n",s);

    float readability = 0.0588 * l - 0.296 * s - 15.8;

    //printf("readability = %f\n", readability);
    //printf("rounded_readability = %i\n", (int) round(readability));

    if (round(readability) < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (round(readability) >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(readability));
    }

// Get Data
    //printf("%i letters \n", letters);
    //printf("%i words \n", words);
    //printf("%i sentences \n", sentences);
}